api/server.py

```diff
@@ -26,27 +26,30 @@
 def run(port, hostname, debug_mode):
     app.run(host=hostname, port=port, debug=debug_mode, threaded=True)
 
-@app.route('/login', methods=["POST"])
+@app.route('/login', methods=["POST", "GET"])
 def login():
-    data = request.get_json()
-    session['username'] = data['username']
-    # TODO: data set and purpose is for now hardcoded, but should be set by the user.
-    session['dataset'] = 'Rotten Tomato'
-    session['purpose'] = 'none'
-    meta_path_loader = MetaPathLoaderDispatcher().get_loader(session['dataset'])
-    meta_paths = meta_path_loader.load_meta_paths()
-    session['meta_path_distributor'] = RandomMetaPathSelector(meta_paths=meta_paths)
-    session['meta_path_id'] = 1
-    session['rated_meta_paths'] = []
+    if request.method == 'POST':
```

Comment:
Why you mix POST and GET here? Couldn't his be handled on two separate methods or does it really do the same stuff?